imagersite/imagersite/views.py

```diff
@@ -14,12 +18,94 @@ def home_view(request):
 
 
 def profile_view(request):
+    user = request.user
+    try:
+        photo_count_pub = user.uphotos.filter(published='pub').count
+        album_count_pub = user.ualbums.filter(published='pub').count
+        # curr_date = str(datetime.datetime.now()).split(' ')[0].split('-')
+        number_of_published_photos = 0
+        for i in user.uphotos.all():
+            x = i.date_published
+            if (datetime.date.today() - x).days < 7:
+                number_of_published_photos += 1
+                # print(curr_date)
+    except AttributeError:
+        return auth_views.login(request)
+
+
     return render(
         request,
         'imagersite/profile.html',
-        context={}
+        context={'user': request.user, 'photo_num': user.uphotos.count, 'album_num': user.ualbums.count, 'album_num_pub':album_count_pub, 'photo_num_pub': photo_count_pub, 'recent': number_of_published_photos}
     )
 
 
+def other_profile_view(request, name):
+    try:
+        user = User.objects.get(username=name)
+        photo_count = user.uphotos.filter(published='pub').count
+        album_count = user.ualbums.filter(published='pub').count
+        return render(
+            request,
+            'imagersite/other_profile.html',
+            context={'user': user, 'photo_num': photo_count, 'album_num': album_count}
+        )
+    except ObjectDoesNotExist:
+        return redirect(home_view)
+
 def account_view(request):
     return render(request, 'registration/registration_form.html')
+
+
+def library_view(request):
+    user = request.user
+    try:
+        photos = user.uphotos.all()
+        albums = user.ualbums.all()
+        return render(request, 'imagersite/library.html', context={'user': user, 'photos': photos, "albums": albums})
+    except AttributeError:
+        return auth_views.login(request)
+
+def photo_view(request, photo_id):
+    photo = list(Photo.objects.all())[int(photo_id)]
```

Comment:
https://docs.djangoproject.com/en/1.11/topics/db/queries/#retrieving-a-single-object-with-get

And ids are indexed from 1 not 0, plus I'm not positive the objects will come out ordered by ids, although they probably will.